src/env/windows.rs

```diff
@@ -0,0 +1,138 @@
+}
+
+/// Deletes an environment variable for the current user.
+fn delete_env_variable(key: &str) -> Result<(), Error> {
+    let root = RegKey::predef(HKEY_CURRENT_USER);
+    let environment = root.open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)?;
+
+    let reg_value = environment.get_raw_value(key);
+    if reg_value.is_err() {
+        return Ok(());
+    }
+
+    env::remove_var(key);
+
+    let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+    let environment_key = hkcu.open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)?;
+    environment_key.delete_value(key)?;
+    Ok(())
+}
+
+/// Sets an environment variable for the current user.
+fn set_env_variable(key: &str, value: &str) -> Result<(), Error> {
+    use std::ptr;
+    use winapi::shared::minwindef::*;
+    use winapi::um::winuser::{
+        SendMessageTimeoutA, HWND_BROADCAST, SMTO_ABORTIFHUNG, WM_SETTINGCHANGE,
+    };
+
+    env::set_var(key, value);
+
+    let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+    let environment_key = hkcu.open_subkey_with_flags("Environment", KEY_WRITE)?;
+    environment_key.set_value(key, &value)?;
+
+    // Tell other processes to update their environment
+    #[allow(clippy::unnecessary_cast)]
+    unsafe {
+        SendMessageTimeoutA(
+            HWND_BROADCAST,
+            WM_SETTINGCHANGE,
+            0 as WPARAM,
+            "Environment\0".as_ptr() as LPARAM,
+            SMTO_ABORTIFHUNG,
+            5000,
+            ptr::null_mut(),
+        );
+    }
+
+    Ok(())
+}
+
+/// Delete the legacy export file.
+fn remove_legacy_export_file() -> Result<(), Error> {
+    let legacy_file = get_home_dir().join(LEGACY_EXPORT_FILE);
+    if legacy_file.exists() {
+        remove_file(&legacy_file)?;
+    }
+
+    Ok(())
+}
+
+// Update the environment for Windows.
+pub(super) fn update_env() -> Result<(), Error> {
+    let mut path = env::var("PATH").unwrap_or_default();
+
+    if let Ok(xtensa_gcc) = env::var("XTENSA_GCC") {
+        let xtensa_gcc: &str = &xtensa_gcc;
+        if !path.contains(xtensa_gcc) {
+            path = format!("{};{}", xtensa_gcc, path);
+        }
+    }
+
+    if let Ok(riscv_gcc) = env::var("RISCV_GCC") {
+        let riscv_gcc: &str = &riscv_gcc;
+        if !path.contains(riscv_gcc) {
+            path = format!("{};{}", riscv_gcc, path);
+        }
+    }
+
+    if let Ok(libclang_path) = env::var("LIBCLANG_PATH") {
+        set_env_variable("LIBCLANG_PATH", &libclang_path)?;
+    }
+
+    if let Ok(libclang_bin_path) = env::var("LIBCLANG_BIN_PATH") {
+        let libclang_bin_path: &str = &libclang_bin_path;
+        if !path.contains(libclang_bin_path) {
+            path = format!("{};{}", libclang_bin_path, path);
+        }
+    }
+
+    if let Ok(clang_path) = env::var("CLANG_PATH") {
+        let clang_path: &str = &clang_path;
+        if !path.contains(clang_path) {
+            path = format!("{};{}", clang_path, path);
+        }
+    }
+
+    set_env_variable("PATH", &path)?;
```

Comment:
I have significant concern about adding some specific versions of Xtensa and RISC-V GCC to the global PATH variable. Before IDF has adopted the "export.sh" approach to managing the environment, we got many bug reports which ended up being due to the compiler version mismatch, for example because the user has been using two versions of IDF (and hence to versions of the compiler) in parallel.

Reply:
Right, there are some scenarios where this could happen, the main scenario being, that we update GCC version. 
 - One possible solution to this could be storing GCC in `$HOME/.rustup/toolchains/esp/<arch>-esp-elf` instead of `$HOME/.rustup/toolchains/esp/<arch>-esp-elf/<gcc_version/<arch>-esp-elf`
   - This would avoid having old GCC version in the user path
   - Note that since the [latest GCC release](https://github.com/espressif/crosstool-NG/releases/tag/esp-13.2.0_20230928), Xtensa got unified GCC under `xtensa-esp-elf`
   
Also, if we are able to use LLD as linker, which should be at some point soon, we don't need to modify user PATH. As:
 - std already downloads (and uses) GCC when building the project
 - no_std only uses GCC as linker, and it would be replaced by LLD
In this case, we only need to set up the `LIBCLANG_PATH` as it's required by std projects.

Reply:
Just a quick note, if we remove the `<gcc_version>` from the installation path, we should check the version when installing (eg using `<target>-gcc --version`)